core/server/apps/subscribers/lib/router.js

```diff
@@ -44,9 +45,13 @@ function honeyPot(req, res, next) {
     next();
 }
 
+function validateUrl(url) {
+    return validator.isEmptyOrURL(url) ? url : '';
+}
+
 function handleSource(req, res, next) {
-    req.body.subscribed_url = req.body.location;
-    req.body.subscribed_referrer = req.body.referrer;
+    req.body.subscribed_url = validateUrl(req.body.location);
+    req.body.subscribed_referrer = validateUrl(req.body.referrer);
```

Comment:
Maybe the function name is wrong, perhaps it should be `ensureURL` instead. 
The validation is handled in validator.isEmptyOrURL, the `validateURL` function is just a wrapper for duplicated code.
If we add middleware for this, that seems like a lot of extra unnecessary code and we then have handling of these values in 2 places.